aeneas/src/vst/Verifier.v3

```diff
@@ -195,6 +195,41 @@ class Verifier(compiler: Compiler, prog: Program) {
 			}
 		}
 	}
+	def checkFieldCycle(decl: VstStruct) {
+		var cv = decl.verifier;
+		for (l = decl.members; l != null; l = l.tail) {
+			var buffer = StringBuilder.new();
+			VstStructField.!(l.head).mtref.render(buffer);
+			var structName = buffer.toString();
+			for (i < cv.file.structs.length) {
+				if (Strings.equal(structName, cv.file.structs[i].name())){
+					if (checkStructCycle(cv.file.structs[i])) {
+						cv.FieldDeclError(l.head, "cyclic field declaration detected");
```

Comment:
Can you inline the error case into the recursive call (i.e. not return a boolean)?

Also, please report the members of the cycle; see the cyclic inheritance check.